src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -496,5 +496,30 @@ internal static string GetMemberCallerName(this Symbol member)
                 member.IsExplicitInterfaceImplementation() ? ExplicitInterfaceHelpers.GetMemberNameWithoutInterfaceName(member.Name) :
                 member.Name;
         }
+
+        /// <summary>
+        /// Locations[0] on lambda symbols covers the entire syntax, which is inconvenient but remains for compatibility.
+        /// For better diagnostics quality, use the NarrowLocation instead, which points to the "delegate" or the "=>".
+        /// </summary>
+        internal static Location NarrowLocation(this MethodSymbol methodSymbol)
```

Comment:
Yes, I understand. I considered it, but felt it was more complicated (new internal API on MethodSymbol and LambdaSymbol).
I don't have a strong preference on this one.